### ISSUE TITLE
ftxvalidator error: "Objective-C garbage collection is no longer supported."

### ISSUE THREAD

**Issue by thundernixon** [googlefonts/fontbakery#2508]
## Observed behaviour

I'm setting up a new computer from scratch, so I'm encoutering some first-time-user fontbakery issues. One of them is this error:

<details>
<summary>💔 <b>ERROR:</b> Checking with ftxvalidator.</summary>

* [com.google.fonts/check/ftxvalidator](https://font-bakery.readthedocs.io/en/latest/fontbakery/profiles/universal.html#com.google.fonts/check/ftxvalidator)
* 💔 **ERROR** ftxvalidator returned an error code. Output follows:

objc[98602]: Objective-C garbage collection is no longer supported.

</details>

I do have ftxvalidator installed, but it seems something is amiss.

## Expected behaviour

I expect ftxvalidator to run without issue.

## Attempts to fix this, so far

[This StackOverflow answer](https://apple.stackexchange.com/a/320961) suggests that I could solve this with:

```
▶ sudo xcodebuild -license accept
```

But this returned:

```
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
```

Based on [this nodejs issue](https://github.com/nodejs/node-gyp/issues/569), it seems that this warning can in turn be fixed with:

```
▶ xcode-select --install # Install Command Line Tools if you haven't already.
▶ sudo xcode-select --switch /Library/Developer/CommandLineTools # Enable command line tools
```

However, for me the first install command results in the message:

```
xcode-select: error: command line tools are already installed, use "Software Update" to install updates
```

So, I found [this SO answer](https://stackoverflow.com/a/34617930) telling me to update.

```
▶ softwareupdate --list
Software Update Tool

Finding available software
Software Update found the following new or updated software:
   * macOS 10.14.5 Update-
	macOS 10.14.5 Update ( ), 2740918K [recommended] [restart]
   * MobileDeviceSU-
	iTunes Device Support Update ( ), 105795K [recommended] [restart]
```

I'll now shut my apps down and try an update to see if that helps, and report back.

## Resources and exact process needed to replicate

I'm working in https://github.com/thundernixon/inter/tree/qa

**Comment by kontur** [comment]
I tried a [new install](https://github.com/googlefonts/fontbakery/issues/2496) of ftxvalidator on Mac OS Mojave 10.14.6 and for me the binaries simply do not work:

```
$ ftxvalidator
objc[1872]: Objective-C garbage collection is no longer supported.
Abort trap: 6
```

Is it time to retire this check entirely if the software from Apple (from 2011) does not execute any more?

**Comment by chrissimpkins** [comment]
The current release of ftxvalidator works for me on Catalina:

```
$ ftxvalidator --version
ftxvalidator build 207, FontToolbox.framework build 335
```

The installation is not as straightforward as it should be with the latest release of ftxvalidator and its associated frameworks https://github.com/googlefonts/fontbakery/issues/2691

**Comment by kontur** [comment]
I eventually resolved that above garbage collection error by removing `/Library/Frameworks/FontToolbox.framework` entirely and reinstalling. 🤷‍♂

**Comment by simoncozens** [comment]
ftxvalidator is no longer used by fontbakery.

**Comment by HinTak** [comment]
FWIW, the "Objective-C garbage collection is no longer supported." error is only seen with the 2011 version (also known as "4 beta 1"). Apple has been doing annual releases of their Font Tools in autumn, sometimes with a beta in the summer. Current is 16 beta (releases in June 2024) , and 15.1 was released in Dec 2023. 

Everything from Font Tools 8 to 16 beta runs on current Mac os. "4 beta 1" is the odd one out.

I compiled a table of all the Apple font tools versions I have gotten my hands on, at:
 https://github.com/FontVal-extras/Apple_Font_Tools_Info
